src/compile.ts

```diff
@@ -11,7 +11,7 @@ import type {
 const functionsStack: FunctionBuildersMap[] = []
 
 export function compile(query: JSONQuery, options?: JSONQueryCompileOptions): Fun {
-  functionsStack.unshift({ ...functions, ...functionsStack[0], ...options?.functions })
+  functionsStack.unshift({ ...functionsStack[0], ...(options?.functions ?? functions) })
```

Comment:
What was the reason for this change? Was it to make the option `functions` be consistent with how `operators` now works?

If so, I prefer the old options (default functions/operators included by default, but you can override them)

Reply:
Right now, the `operators` options changes to require passing all operators instead of only your custom operators. For consistency, I think it will be good to do the same with the `functions` option.

If we change the `operators` API again to something that does not require passing all operators, we can reverse this change in `functions`.